jgrapht-core/src/test/java/org/jgrapht/GraphsTest.java

```diff
@@ -455,6 +458,1255 @@ public void vertexHasParents_A()
+
+        Assert.assertEquals(null, edge);
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void addEdgeWithInvalidVertexTest() {
+        SimpleGraph<String, DefaultWeightedEdge> graph = new SimpleGraph<>(new ClassBasedEdgeFactory<String, DefaultWeightedEdge>(DefaultWeightedEdge.class), true);
+
+        String a = "A";
+        String b = "B";
+        String c = "C";
+        final double weigth = 10.0;
+
+        graph.addVertex(a);
+        graph.addVertex(b);
+
+        Graphs.<String, DefaultWeightedEdge>addEdge(graph, a, c, weigth);
+    }
+
+    /*
+     *  Tests for public static <V, E> E addEdgeWithVertices(Graph<V, E> g, V sourceVertex, V targetVertex) 
+     */
+    @Test
+    public void addEdgeWithVerticesExistingVerticesTest() {
+        SimpleGraph<String, DefaultEdge> graph = new SimpleGraph<>(DefaultEdge.class);
+
+        String a = "A";
+        String b = "B";
+
+        graph.addVertex(a);
+        graph.addVertex(b);
+
+        DefaultEdge edge = Graphs.<String, DefaultEdge>addEdgeWithVertices(graph, a, b);
+
+        Assert.assertNotEquals(null, edge);
+        Assert.assertTrue(graph.getEdgeSource(edge).equals(a));
+        Assert.assertTrue(graph.getEdgeTarget(edge).equals(b));
+    }
+
+    @Test
+    public void addEdgeWithVerticesNewVerticesTest() {
+        SimpleGraph<String, DefaultEdge> graph = new SimpleGraph<>(DefaultEdge.class);
+
+        String a = "A";
+        String b = "B";
+
+        DefaultEdge edge = Graphs.<String, DefaultEdge>addEdgeWithVertices(graph, a, b);
+
+        Assert.assertNotEquals(null, edge);
+        Assert.assertTrue(graph.getEdgeSource(edge).equals(a));
+        Assert.assertTrue(graph.getEdgeTarget(edge).equals(b));
+    }
+
+    /*
+     *  Tests for public static <V,
+     *      E> boolean addEdgeWithVertices(Graph<V, E> targetGraph, Graph<V, E> sourceGraph, E edge)
+     */
+    @Test
+    public void addEdgeWithVertices2ExistingVerticesTest() {
+        SimpleGraph<String, DefaultEdge> graphSource = new SimpleGraph<>(DefaultEdge.class);
+        SimpleGraph<String, DefaultEdge> graphTarget = new SimpleGraph<>(DefaultEdge.class);
+
+        String a = "A";
+        String b = "B";
+
+        graphSource.addVertex(a);
+        graphSource.addVertex(b);
+
+        graphTarget.addVertex(a);
+        graphTarget.addVertex(b);
+
+        DefaultEdge edgeSource = Graphs.<String, DefaultEdge>addEdgeWithVertices(graphSource, a, b);
+
+        Boolean result = Graphs.<String, DefaultEdge>addEdgeWithVertices(graphTarget, graphSource, edgeSource);
+
+        DefaultEdge edgeTarget = graphTarget.getEdge(a, b);
+        edgeSource = graphSource.getEdge(a, b);
+
+        Assert.assertTrue(result);
+        Assert.assertTrue(graphTarget.getEdgeSource(edgeTarget).equals(a));
+        Assert.assertTrue(graphTarget.getEdgeTarget(edgeTarget).equals(b));
+        Assert.assertTrue(graphSource.getEdgeSource(edgeSource).equals(a));
```

Comment:
No need to test this. This should be covered by tests for the method addEdgeWithVertices(graph, sourceVertex, targetVertex)